### ISSUE TITLE
Philhk Vencord Fork is not working because vencord stopped working and stuff needs to be updated

### ISSUE THREAD

**Issue by Pickymarker** [philhk/Vencord#24]
### What happens when the bug or crash occurs?

Philhk Vencord Fork is not working because vencord stopped working and stuff needs to be updated

### What is the expected behaviour?

Philhk Vencord Fork is not working because vencord stopped working and stuff needs to be updated

### How do you recreate this bug or crash?

Philhk Vencord Fork is not working because vencord stopped working and stuff needs to be updated

### Errors

Philhk Vencord Fork is not working because vencord stopped working and stuff needs to be updated

### Request Agreement

- [X] I am using Discord Stable or tried on Stable and this bug happens there as well
- [X] I have read the requirements for opening an issue above

**Comment by Lioncat6** [comment]
Just build it yourself until they merge it. Just download this repo, copy the plugins and paste them in a download of the main vencord repo.

**Comment by Lioncat6** [comment]
`philsPluginLibrary`
`betterMicrophone.desktop`
`betterScreenshare.desktop` 
are the plugins that you should copy

**Comment by BlueberryWolf** [comment]
you can also use the easy install script on my profile, it merges and builds everything for you

**Comment by Lioncat6** [comment]
Neat! Unfortunately I can't use it cause my path variable is fucked lol.

**Comment by Killua81** [comment]
betterMicrophone.desktop and betterScreenshare.desktop they don't appear the way they used to. the icons are gone and bettermicrophone no longer works.

I try to use @BlueberryWolf launcher and this is the same.

**Comment by BlueberryWolf** [comment]
> betterMicrophone.desktop and betterScreenshare.desktop they don't appear the way they used to. the icons are gone and bettermicrophone no longer works.
> 
> I try to use @BlueberryWolf launcher and this is the same.

Actually, the plugin never broke, only the UI. I will look into it at some point, I'm just not feeling at it at the moment.

If you still want to use the plugin, you can manually edit the settings json to change your mic and screenshare settings, as only the UI is broken

**Comment by Annabxlla** [comment]
Screenshare fully crashes vencord now, and the settings json for better mic has all of its profiles deleted, including the default ones it creates.

Also NTTS made a [video](https://www.youtube.com/watch?v=_rCXxa5MDrE) about this plugin so discord might patch betterScreenshare

**Comment by Pickymarker** [comment]
wondering if yadb4nitro that was worked on to get working with better discord compat for vencord plugin actually works and can be used instead of this because yabdp4nitro works on better discord and was supposedly made to work with that plugin

**Comment by BlueberryWolf** [comment]
> Screenshare fully crashes vencord now, and the settings json for better mic has all of its profiles deleted, including the default ones it creates.
> 
> Also NTTS made a [video](https://www.youtube.com/watch?v=_rCXxa5MDrE) about this plugin so discord might patch betterScreenshare

BetterMicrophone still works, and you can even get the configurator to open if you call the method directly. the only thing broken is the UI. I am planning to fix it soon but I cannot at the moment, as my neighbours house burned down and destroyed my PC peripherals. I can't afford a mouse at the moment after replacing other stuff so, I can't really do a whole lot to fix it yet

**Comment by qamwxy** [comment]
yeah, now it crashes instantly.. hmmm

**Comment by qamwxy** [comment]
> I am planning to fix it

Do you really think its only the UI? Would be strange if it suddenly broke down, because I don't think the UI components and requirements change that often

**Comment by qamwxy** [comment]
> Also NTTS made a [video](https://www.youtube.com/watch?v=_rCXxa5MDrE) about this plugin so discord might patch betterScreenshare

Strange it works for him but not for anyone else

**Comment by qamwxy** [comment]
> If you still want to use the plugin, you can manually edit the settings json to change your mic and screenshare settings, as only the UI is broken

you can't, the button to stream instantly crashes your discord and right click doesn't open anything

**Comment by BlueberryWolf** [comment]
> > If you still want to use the plugin, you can manually edit the settings json to change your mic and screenshare settings, as only the UI is broken
> 
> you can't, the button to stream instantly crashes your discord and right click doesn't open anything

that comment was from 5 days ago bro, the crashes are more recent than that. BetterScreenshare is toast rn, but BetterMicrophone still works without the UI (you can actually get the microphone settings to open if you patch index.js to expose the function)

**Comment by Loukious** [comment]
I fixed the issues in [my fork](https://github.com/Loukious/Vencord) if anyone is interested.
You can install the fork by using my Vencord [installer](https://github.com/Loukious/Installer/releases) fork. 
It's mostly for my personal use and provided as is and there's no guarantee that I'll push future updates when it breaks.
Enjoy!

**Comment by 4kaytrey** [comment]
> I fixed the issues in [my fork](https://github.com/Loukious/Vencord) if anyone is interested. You can install the fork by using my Vencord [installer](https://github.com/Loukious/Installer/releases) fork. It's mostly for my personal use and provided as is and there's no guarantee that I'll push future updates when it breaks. Enjoy!

I installed the fork which you made and it still seems to crash when you join calls, or itll just say RTC Connecting.

**Comment by Loukious** [comment]
> > I fixed the issues in [my fork](https://github.com/Loukious/Vencord) if anyone is interested. You can install the fork by using my Vencord [installer](https://github.com/Loukious/Installer/releases) fork. It's mostly for my personal use and provided as is and there's no guarantee that I'll push future updates when it breaks. Enjoy!
> 
> I installed the fork which you made and it still seems to crash when you join calls, or itll just say RTC Connecting.

Should probably check your plugin list. Mine is working perfectly. 
![image](https://github.com/user-attachments/assets/921960e0-c69c-4c15-8e08-bff18c8ef49a)
![image](https://github.com/user-attachments/assets/aefda158-52ed-42e0-b389-03d0190f7b8f)
![image](https://github.com/user-attachments/assets/25a9b170-e167-4617-85a9-80e0751d1ce6)
![image](https://github.com/user-attachments/assets/7f2a21c5-6f6c-43fb-9950-247b63804ea3)
Posting logs from console also help.

**Comment by qamwxy** [comment]
very sketchy, wouldn't trust that installer at all:
danger 3
T1552.001 Credentials In Files (1)
Actions looks like stealing of personal data
T1518 Software Discovery (1)
Actions looks like stealing of personal data
Drops the executable file immediately after the start

**Comment by qamwxy** [comment]
nvm the official one has the same detections

**Comment by qamwxy** [comment]
I mean it works, sketchy af tho

and buggy

**Comment by Annabxlla** [comment]
I never posted a comment about it but I already took a look at the commits, and it looks fine.

**Comment by qamwxy** [comment]
![image](https://github.com/user-attachments/assets/ae8863e3-9067-4a55-8d0f-8f0bfe6b6352)
![image](https://github.com/user-attachments/assets/d2ab9eac-9b62-46b2-a8a6-c4d273a9a837)
![image](https://github.com/user-attachments/assets/16f8cea3-0cda-4bbc-b835-81a4051d3036)
just essentials plugins for me, still manages to infinitely rtc connect

**Comment by qamwxy** [comment]
same thing with only that plugin

**Comment by qamwxy** [comment]
Issue resolved, setup your betterscreenshare settings first
(The fork client issue)

Finally, 240FPS stream and custom audio sorce

PS. Doesn't look like 240FPS, but 60

It doesn't change the fps, only the number that it shows, 99999999 fps doesn't crash anymore

**Comment by philhk** [comment]
Unfortunately the framerate is currently broken and I'm unsure of the cause. If anyone discovers a fix please feel free to submit a pull request. However I would advise against spending too much time on this repository. If you are really motivated I recommend creating a whole new plugin from scratch using better practices than those used here.

**Comment by Annabxlla** [comment]
It does change what fps the other users see, and after testing with 1 fps and such it does seem to work, I don't have a way to test higher fps amounts as I'm not home and only have my phone and laptop with me.

**Comment by philhk** [comment]
Yeah it's quite weird. You can set any custom frame rate between 1 and 30, but you can't go above that.

**Comment by BlueberryWolf** [comment]
> Yeah it's quite weird. You can set any custom frame rate between 1 and 30, but you can't go above that.

it's because the client sets the requesting framerate. you can send a higher framerate stream but the client will not request frames at that rate. You would have to patch both the sending and receiving clients

**Comment by Tresillo2017** [comment]
I've tested and using the codec AV1 in the debug panel frame rate will go up to 120fps. Seems to be limited due to the codec for lower bandwidth.

**Comment by Tether1776** [comment]
Has anyone gotten the HDR streaming to work. I can't figure it out.

**Comment by Killua81** [comment]
UI for better screenshare and microphone is broken....again
